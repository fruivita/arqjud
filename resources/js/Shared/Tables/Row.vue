<!--
    Linha (tr) da tabela padrÃ£o.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<template>
    <tr class="even:bg-primaria-100 dark:even:bg-secundaria-800" dusk="slot">
        <slot></slot>
    </tr>
</template>
