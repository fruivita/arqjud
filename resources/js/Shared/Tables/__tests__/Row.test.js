/**
 * Testes para o componente Row.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Row from '@/Shared/Tables/Row.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(Row, { ...options });
    };
});

describe('Row', () => {
    // Caminho feliz
    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { default: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });
});
