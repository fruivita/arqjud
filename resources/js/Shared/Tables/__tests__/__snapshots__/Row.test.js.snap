// Vitest Snapshot v1

exports[`Row > renderiza o componente respeitando o snapshot 1`] = `"<tr class=\\"even:bg-primaria-100 dark:even:bg-secundaria-800\\" dusk=\\"slot\\">foo</tr>"`;
